chichiwang/exercise-js
# Build Home page section: Usage

## Description
Linked to Issue: #64
Build the Home page section "Usage" that links to the `/documentation` page. Additionally, add a bare-bones `/documentation` page to the router, and a `Documentation` component to render in the new route. The page link to `Counter` in the Nav is replaced with a page link to `Documentation` instead.

A new component `ContentBlock` is created to constrain the width of a page's content, as well as center it. The route `/counter` and page component `Counter` are removed as they are no longer used.

## Changes
* `app/dataRoutes/index.ts` is updated
  * Route `/counter` removed
  * Route `/documentation` added to render the `pages/Documentation` component
* `pages/Counter` directory is removed as it is no longer used
* A new page component `pages/Documentation/index.tsx` is created
  * Bare-bones placeholder content is rendered
* `pages/Home/index.tsx` is updated
  * A new `HeadingBlock` "Usage" containing a link to the `/documentation` page
  * All content below the `HeroBanner` is wrapped in a `ContentBlock` component
* `pages/Layout/Nav.tsx` is updated
  * Link to `/counter` is removed and replaced with a link to `/documentation`
* A `pages/components/ContentBlock/index.tsx` component is created
  * Constrains the content width for a page to `1200px` and horizonatally centers the content within whatever container it is placed in
* `pages/components/HeadingBlock/styles.module.scss` is updated to set the width of the HeadingBlock to `100%` and remove horizontal margin-centering

## Steps to QA
* Pull down and switch to this branch
* Run the app and verify:
  * Home page looks the same as before (on `main branch`)
  * Home page contains a new "Usage" section and the link in the section to `/documentation` works correctly
  * Link in the Nav bar to `Documentation` works correctly
  * Routing directly to (or refreshing on) `/documentation` works correctly
